File: python-library/container/container.py
from abc import ABC, abstractmethod
from typing import List
from .container_result import ContainerResult


class Container(ABC):
    """Container interface for tag management"""

    @abstractmethod
    def create(self) -> ContainerResult:
        """Create container with default tags"""
        pass

    @abstractmethod
    def add(self, tag: 'Tag') -> ContainerResult:
        """Add tag to container"""
        pass

    @abstractmethod
    def remove(self, tag: 'Tag') -> ContainerResult:
        """Remove tag from container"""
        pass

    @abstractmethod
    def build(self) -> ContainerResult:
        """Build ordered list of tags"""
        pass

    @abstractmethod
    def content(self) -> ContainerResult:
        """Export container content as bytes"""
        pass

File: kotlin-library/src/main/kotlin/container/Container.kt
package container

import tag.Tag

interface Container {
    fun create(): ContainerResult<Unit>
    fun add(tag: Tag): ContainerResult<Unit>
    fun remove(tag: Tag): ContainerResult<Unit>
    fun build(): ContainerResult<List<Tag>>
    fun content(): ContainerResult<ByteArray>
}